cranelift/codegen/src/remove_constant_phis.rs

```diff
@@ -112,6 +111,9 @@ impl AbstractValue {
 struct OutEdge<'a> {
     /// An instruction that transfers control.
     inst: Inst,
+    /// The index into branch_destinations for this instruction that corresponds
+    /// to this edge.
+    branch_index: u8,
```

Comment:
Will this always be enough? I don't remember if we have any explicitly-enforced implementation limits on `br_table` successor count, but we should verify this (and leave a comment here, and possibly enforce in the validator) if we stick with a `u8`.

Reply:
`cranelift_frontend::Switch` assumes that everything that fits in 32bits works. It is definitively realistic to have more than 256 entires in a `br_table` for rust code.

Reply:
Good call, I'll bump this to `u32`.